src/Time/TimeId.hpp

```diff
@@ -69,7 +69,7 @@ class TimeId {
   void canonicalize() noexcept;
 
   bool time_runs_forward_{false};
-  size_t slab_number_{0};
+  ssize_t slab_number_{0};
```

Comment:
- I think in the initialization you are assuming slab_number can contain "large" negative values, but according to the POSIX standard it is only garuanteed to be capable of storing values at least in the range [-1, {SSIZE_MAX}]. Maybe it would be better to use int or an extra variable for the initialization.
- the POSIX header <sys/types.h> is missing

Reply:
- I think you've misread something.  What you call "large" here is about -7.

- We (by which I mostly mean "I") apparently decided cstddef was sufficient.  I have no idea why, since that looks completely wrong and we should have sys/types.h as you say.  I'll fix that.

Reply:
Yes, there is no correct type for this, since the whole idea of size_t is that it specifies a (positive) size of sth.. I think the cleanest and more transparent way would be using a separate variable for the initialization.

Reply:
I'm with @wthrowe on this. I like having self-starting just being negative values. Whether we should use `ssize_t`, `std::ptrdiff_t`, or `int64_t` for signed-size_t-like things should maybe be discussed in an issue.

Reply:
In this particular case I've decided I like `int64_t` better anyway, so I'll do that.  At some point I'll also look at a few of my files that should probably be using `ptrdiff_t` or `some_random_thing::difference_type` anyway.